index.js

```diff
@@ -40,6 +42,25 @@ function increaseSpecifityOfRule(rule, opts) {
 	}
 }
 
+/**
+ * Checks for the presence of enable/disable comments up to the previous rule.
+ * @param  {node} node 		PostCSS Node
+ * @return {Null|Boolean} Null if no matching comment; otherwise boolean (toggle status)
+ */
+function checkForEnableDisableComments(node) {
+	if (!node) return null;
+
+	for (var prev = node; prev = prev.prev(); ) {
+		if (prev.type !== 'comment') return null;
+		if (DISABLE_PLUGIN_RE.test(prev.text)) {
+			return true;
+		} else if (ENABLE_PLUGIN_RE.test(prev.text)) {
+			return false;
+		}
+	}
+
+	return null;
+}
```

Comment:
This function currently checks from the given node upwards through any comments. But there can be other types of nodes in the way. Let's be sure to add a test

```css
/* postcss-increase-specificity disable */

random-decl-here: 'why';

.foo {
	background: #f00;
}
/* postcss-increase-specificity enable */

.bar {
	color: #ef0;
}
```

I still think we should `checkForEnableDisableComments(fromNode, toNode)` and `toNode.prev(...)` loop to `toNode`, stopping at the first matching disable/enable comment.

Reply:
@MadLittleMods There is no genuine use case for a declaration outside of a class; that's not valid CSS.

Reply:
@track0x1 Just because it isn't normal CSS, doesn't mean that it come up as it parses fine with PostCSS. We should handle this situation.

Reply:
Ok, so I think we can safely ignore declarations.

Reply:
This logic is still flawed. Consider the following example (let's add a test as well),

```css
/* postcss-increase-specificity disable */

.foo {
	background: #f00;
}
/* postcss-increase-specificity enable */

@some-at-rule {

}

.bar {
	color: #ef0;
}
```

---

Please update this method to be generic to handle any type of node in between, even future unknown types.

I think we should go with this,

```js
/**
 * Checks for the presence of enable/disable comments
 * starting from the `fromNode` upwards to the previous rule (`toNode`).
 * @param  {fromNode} node 		PostCSS Node
 * @param  {toNode} node 		PostCSS Node
 * @return {Null|Boolean} Null if no matching comment; otherwise boolean (toggle status)
 */
function checkUpwardsForEnableDisableComments(fromNode, toNode) {
	if (!fromNode) return null;

	for (var prev = fromNode.prev(); (prev !== toNode && prev !== undefined); prev = prev.prev()) {
		if (DISABLE_PLUGIN_RE.test(prev.text)) {
			return true;
		} else if (ENABLE_PLUGIN_RE.test(prev.text)) {
			return false;
		}
	}

	return null;
}
```